include/seastar/core/file-types.hh

```diff
@@ -158,6 +159,18 @@ inline constexpr file_permissions operator&(file_permissions a, file_permissions
     return file_permissions(std::underlying_type_t<file_permissions>(a) & std::underlying_type_t<file_permissions>(b));
 }
 
+/// seastar::space_info is equivalent to std::filesystem::space_info
+/// with renamed members, to prevent a conflict with future::available()
+struct space_info {
```

Comment:
Why not use std::filesystem::space_info?

Reply:
I mentioned it in a comment.
The compiler complains about it due to the `available` member conflicting with `future::available`.  I can reproduce and quote the error

Reply:
How can it conflict with future::available? It's in a different class.

Reply:
I see the problem.

Reply:
https://github.com/scylladb/seastar/pull/2598

Reply:
v2 is rebased onto #2598 for now